mlflow/openai/utils.py

```diff
@@ -99,3 +101,49 @@ def request(*args, **kwargs):
             return original(*args, **kwargs)
 
     return _mock_request(new=request)
+
+
+class _OAITokenHolder:
+    def __init__(self, api_type):
+        self.api_type = api_type
+        self._api_token = None
+        self._credential = None
+
+        if self.api_type in ("azure_ad", "azuread"):
```

Comment:
```suggestion
        self._is_azure_ad = self.api_type in ("azure_ad", "azuread")

        if self._is_azure_ad:
```
We can avoid the duplicated conditional check for api type :) [line 128]